src/test/java/com/beust/jcommander/JCommanderTest.java

```diff
@@ -348,6 +359,9 @@ public void hiddenSplitter() {
       @Parameter(names = "--extensions", splitter = HiddenParameterSplitter.class)
       public List<String> extensions;
     }
+    if (HiddenParameterSplitter.class.getConstructors().length == 0) {
+      return; // Compiler has optimised away the private constructor
```

Comment:
Someone else also pointed that out to me and I meant to write another patch
for that but never did so.  The reason I made that mistake is that the code
I am testing calls getConstructor(), hitting the same problem

On 5 September 2015 at 08:48, Anton Gustafsson <notifications@github.com>
wrote:

> In src/test/java/com/beust/jcommander/JCommanderTest.java
> <https://github.com/cbeust/jcommander/pull/243#discussion_r38808689>:
>
> > @@ -348,6 +359,9 @@ public void hiddenSplitter() {
> >        @Parameter(names = "--extensions", splitter = HiddenParameterSplitter.class)
> >        public List<String> extensions;
> >      }
> > +    if (HiddenParameterSplitter.class.getConstructors().length == 0) {
> > +      return; // Compiler has optimised away the private constructor
>
> Class.getConstructors() does not include private constructors, so the
> returned array will always have a length of zero. The java compiler *does
> not* optimize away much; mostly inlining primitive final fields and
> eliminating dead code. Other
> optimizations are performed at runtime by the JVM.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cbeust/jcommander/pull/243/files#r38808689>.
>